src/components/projects-summary/projects-summary.controller.ts

```diff
@@ -1,23 +1,144 @@
+    this.DataService = DataService;
+    this.AlertMessageService = AlertMessageService;
   }
 
-  public closePanel = () => {
-    return;
+  public $onInit () {
+    this.ProjectsService.canCreate().then(() => {
+      this.ctrl.canCreate = true;
+    }, (result) => {
+      this.ctrl.canCreate = false;
+
+      var data = result.data || {};
+
+      // 403 Forbidden indicates the user doesn't have authority.
+      // Any other failure status is an unexpected error.
+      if (result.status !== 403) {
+        var msg = 'Failed to determine create project permission';
+        if (result.status !== 0) {
+          msg += " (" + result.status + ")";
+        }
+        this.Logger.warn(msg);
+        return;
+      }
+
+      // Check if there are detailed messages. If there are, show them instead of our default message.
+      if (data.details) {
+        var messages = [];
+        _.forEach(data.details.causes || [], function (cause: any) {
+          if (cause.message) {
+            messages.push(cause.message);
+          }
+        });
+        if (messages.length > 0) {
+          this.ctrl.newProjectMessage = messages.join("\n");
+          // TODO: Toast Notification
+        }
+      }
+    });
+
+    this.AuthService.withUser().then(this.onVerifyUser);
+  }
+
+  public onVerifyUser = () => {
+    this.ctrl.loading = true;
+    this.watches.push(this.DataService.watch('projects', this.$scope, this.onProjectsUpdate));
+    this.AlertMessageService.getAlerts().forEach(function(alert: any) {
+      this.ctrl.alerts[alert.name] = alert.data;
+    });
+  };
+
+  public onProjectsUpdate = (projectData: any) => {
+    var projects: any = _.toArray(projectData.by('metadata.creationTimestamp'));
+    var orderByDate: any = this.$filter('orderObjectsByDate');
+    this.ctrl.projects = orderByDate(projects, true);
+
+    this.ctrl.maxDisplayProjectsReached = this.ctrl.projects.length > this.maxDisplayProjects;
+    this.ctrl.totalProjects = this.ctrl.projects.length;
+    if (this.ctrl.maxDisplayProjectsReached) {
+      this.ctrl.projects.splice(this.maxDisplayProjects, this.ctrl.projects.length);
+    }
+    this.ctrl.loading = false;
+    this.ctrl.showGetStarted = _.isEmpty(this.ctrl.projects);
+  };
+
+  public openNewProjectPanel() {
+    this.ctrl.showNewProjectPanel = true;
+  }
+
+  public closeNewProjectPanel = () => {
+    this.ctrl.showNewProjectPanel = false;
   };
-}
 
+  public onNewProject = (projectName: string) => {
+    this.ctrl.showNewProjectPanel = false;
+  };
+
+  public editProject = (project: any, parentId: string) => {
+    var projectDiv = jQuery(this.$element).find('#' + parentId)[0];
+    var listDiv = jQuery(this.$element).find('#catalogs-projects-summary-list')[0];
+    var width = this.$window.innerWidth;
+    var w: any = jQuery(window)[0];
+
+    if (width > w.patternfly.pfBreakpoints.tablet) {
+      this.ctrl.editPanelTopPosition = listDiv.offsetTop + projectDiv.offsetTop + projectDiv.clientHeight - 10;
```

Comment:
We could if we added the editPanel to each project tile (similar to uib dropdowns). This implementation requires only 1 panel that gets placed based on the project being edited.